lib/common/models/concerns/maximum_redis_lifetime.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+require 'common/models/redis_store'
+
+module Common
+  module MaximumRedisLifetime
+    extend ActiveSupport::Concern
+
+    included do
+      unless self < Common::RedisStore
+        raise ArgumentError, 'Class composing Common::MaximumRedisLifetime must be a Common::RedisStore'
+      end
+
+      extend(ClassMethods)
+
+      validate :within_maximum_ttl
+      validates :created_at, presence: true
```

Comment:
This is probably good enough, but seems like it is something that should checked when the mixin is added similar to the ArgumentError above. It's kind of weird that we check for it in an instance.